src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/Microsoft.CommonLanguageServerProtocol.Framework.Shared.projitems

```diff
@@ -0,0 +1,45 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <MSBuildAllProjects Condition="'$(MSBuildVersion)' == '' Or '$(MSBuildVersion)' &lt; '16.0'">$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+    <HasSharedItems>true</HasSharedItems>
+    <SharedGUID>64eaded3-4b5d-4431-bbe5-a4aba1c38c00</SharedGUID>
+  </PropertyGroup>
+  <PropertyGroup Label="Configuration">
+    <Import_RootNamespace>Microsoft.CommonLanguageServerProtocol.Framework.Shared</Import_RootNamespace>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="$(MSBuildThisFileDirectory)AbstractHandlerProvider.cs" />
```

Comment:
>AbstractHandlerProvider.cs [](http://example.com/codeflow?start=49&length=26)

Out of curiosity, why not an all-inclusive glob?

Reply:
Was autogenerated by VS - also seems to match what other shared projects do (probably because they were also autogenerated)
https://github.com/dotnet/roslyn/blob/main/src/Dependencies/PooledObjects/Microsoft.CodeAnalysis.PooledObjects.projitems